### PR TITLE
Improve type inference for plugin managers, Drop Interop\Container, Drop PHP 7.3

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | yes

### Description

- Drop PHP 7.3
- Removes Interop\Container via ServiceManger 3.14 minimum
- Makes `ExceptionInterface` extend from `Throwable`
- Re-enables translation-related tests that depend on ServiceManager ^3
- Adds templated types to the `RoutePluginManager`
- Deprecates ServiceManager v2 factory methods
- Other minor improvements

### CODE REVIEW

File: src/ConfigProvider.php [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 namespace Laminas\Router;
 
-use Zend\Router\Http\TreeRouteStack;
+use Laminas\ServiceManager\ConfigInterface;
```
Comment (Ocramius) [inline src/ConfigProvider.php:7]: What a find! 😱

File: src/Exception/ExceptionInterface.php [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 namespace Laminas\Router\Exception;
 
-interface ExceptionInterface
+use Throwable;
+
+interface ExceptionInterface extends Throwable
```
Comment (Ocramius) [inline src/Exception/ExceptionInterface.php:9]: Possibly a BC break, but 🤷‍♀️

File: src/RoutePluginManager.php [hunk 3]
```diff
@@ -24,13 +25,19 @@
...
+ * @extends AbstractPluginManager<InstanceType>
+ * @psalm-import-type ServiceManagerConfiguration from ServiceManager
  */
 class RoutePluginManager extends AbstractPluginManager
 {
     /**
      * Only RouteInterface instances are valid
      *
-     * @var string
+     * @var class-string
```
Comment (Ocramius) [inline src/RoutePluginManager.php:40]: not important, but consider that `interface-string` also exists (AFAIK)

**Review by Ocramius — APPROVED** [review]

File: src/RoutePluginManager.php (hunk 3 above)
Comment (gsteel) [reply src/RoutePluginManager.php:40]: Oooh! Didn't know that.

**Pull request merged**